### PR TITLE
In-Context Editor Chrome Extension

### PR DESCRIPTION
**Pull request by sunloverz**

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+chrome.webRequest.onHeadersReceived.addListener(
+    function (details) {
+        for (var i = 0; i < details.responseHeaders.length; ++i) {
+            if (details.responseHeaders[i].name.toLowerCase() == 'x-frame-options') {
+                details.responseHeaders.splice(i, 1);
```
Comment (dmitriykharchenko) [inline background.js]: Here is a buggy code, you shouldn't splice an array that you iterating over.

Comment (dmitriykharchenko) [inline background.js]: Chrome supports es6, might be a good idea to use `map`/`reduce` methods instead.

File: popup.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// var app = chrome.runtime.getBackgroundPage();
+
+
+document.body.onload = function() {
+    chrome.storage.sync.get("isEnabled", function(items) {
+        if (!chrome.runtime.error && items.isEnabled == true) {
```
Comment (dmitriykharchenko) [inline popup.js]: `==` — a bad practice to use this thing to compare

File: popup.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+               chrome.runtime.sendMessage({popupOpen: true, url: url});
+            });
+        }
+    });
+};
+
+document.getElementById("agree-btn").onclick = function() {
+    var isEnabled = document.getElementById("agree").checked;
+    chrome.storage.sync.set({ "isEnabled" : isEnabled }, function() {
+        if (chrome.runtime.error) {
```
Comment (dmitriykharchenko) [inline popup.js]: Might be a good idea to handle this error, to make the extension a bit more crash-resistant.

Comment (sunloverz) [inline popup.js]: Its just callback on success, or on failure (in which case runtime.lastError will be set)

**Comment by dmitriykharchenko** [comment]
LGTM

**Comment by jonathanpeterwu** [comment]
@dmitriykharchenko  @paton this PR is ready to go and tested, will need it merged into master so I can upload it to the chrome extension store

**Comment by jonathanpeterwu** [comment]
@paton  circling back as this is a blocker for chrome extension

File: popup.js (hunk 3 above)
Comment (jonathanpeterwu) [reply popup.js]: @dmitriykharchenko  can you complete this change

**Pull request merged**